From 8e7394d715d846937168056576e98658a195bf40 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Fri, 19 Feb 2016 12:44:22 +0800
Subject: [PATCH 59/65] serial: imx: hack to fix suspend/resume

This patch was NAKed by rmk, but it seems to do the job, so keep it
around for now until a better solution is provided.

Signed-off-by: Sean Cross <xobs@kosagi.com>
---
 drivers/tty/serial/imx.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 016e4be..16a551b 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -2071,13 +2071,13 @@ static int imx_serial_port_suspend_noirq(struct device *dev)
 	struct imx_port *sport = platform_get_drvdata(pdev);
 	int ret;
 
-	ret = clk_enable(sport->clk_ipg);
+	ret = clk_prepare_enable(sport->clk_ipg);
 	if (ret)
 		return ret;
 
 	serial_imx_save_context(sport);
 
-	clk_disable(sport->clk_ipg);
+	clk_disable_unprepare(sport->clk_ipg);
 
 	return 0;
 }
@@ -2088,13 +2088,13 @@ static int imx_serial_port_resume_noirq(struct device *dev)
 	struct imx_port *sport = platform_get_drvdata(pdev);
 	int ret;
 
-	ret = clk_enable(sport->clk_ipg);
+	ret = clk_prepare_enable(sport->clk_ipg);
 	if (ret)
 		return ret;
 
 	serial_imx_restore_context(sport);
 
-	clk_disable(sport->clk_ipg);
+	clk_disable_unprepare(sport->clk_ipg);
 
 	return 0;
 }
-- 
2.7.3

